src/pages/RestrictedAction/Workspace/WorkspaceUserRestrictedAction.tsx

```diff
@@ -39,7 +38,7 @@ function WorkspaceUserRestrictedAction({policyID}: WorkspaceUserRestrictedAction
         >
             <HeaderWithBackButton
                 title={translate('workspace.restrictedAction.restricted')}
-                onBackButtonPress={Navigation.goBack}
+                onBackButtonPress={Navigation.resetToHome}
```

Comment:
On returning from the Restricted Actions view for users, why do we need to reset the navigation history? Seems too harsh on the user for no fault of his. 

Personally, I would not recommend to reset the navigation history even for Admins and Owner. Also, we reset the navigation history when we guide the user/admin/owner to the appropriate report. 

And, one last point. For the scenario of manual/split request, I think the user would expect to go back to the participant selection page instead of abruptly ending the request. no?

Is this all intentional/agreed upon somewhere? Maybe I am missing something here.

Reply:
If I keep the previous behavior this is what happens on mobile for example:


https://github.com/Expensify/App/assets/20051562/0f3c97df-6f73-43b1-848b-5cdaec150603

FAB -> Submit expense -> Restricted Action -> Workspace chat

If I go back on the workspace chat I will return to the restricted action, which i don't think it makes much sense. I'm not familiar with the navigation system so that's why I'm resetting like this, do you have any thoughts about the flow with goBack vs resetToHome?

Reply:
I think the comment [here](https://github.com/Expensify/App/pull/44393#discussion_r1659254359) may solve this problem.